### PR TITLE
[BEAM-1988] Add join operation to the filesystem

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

R: @chamikaramj PTAL

### CODE REVIEW

**Comment by sb2nov** [comment]
R: @chamikaramj should be ready for review

**Comment by chamikaramj** [comment]
LGTM

File: sdks/python/apache_beam/io/gcp/gcsfilesystem_test.py [hunk 1]
```diff
@@ -36,6 +36,15 @@
 @unittest.skipIf(gcsfilesystem is None, 'GCP dependencies are not installed')
 class GCSFileSystemTest(unittest.TestCase):
 
+  def test_join(self):
```
Comment (aaltay) [inline sdks/python/apache_beam/io/gcp/gcsfilesystem_test.py]: Could you add a test where the first argument ends with `/`

Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/gcsfilesystem_test.py]: Got it, will do.

**Pull request closed without merging**